tests/test_diff.cpp

```diff
@@ -155,7 +153,7 @@ TEST_CASE("laplacian", "[test_adjoint]")
 	Eigen::MatrixXd velocity_discrete;
 	sample_field(*states[0], velocity, velocity_discrete);
 
-	verify_adjoint(*nl_problem, x, velocity_discrete, 1e-7, 3e-5);
+	verify_adjoint(*nl_problem, x, velocity_discrete, 1e-7, 1e-8);
```

Comment:
nice! why the change?

Reply:
there was a bug in laplacian, it's a miracle that it's been there for two years.